src/libs/actions/App.js

```diff
@@ -177,33 +170,31 @@ function fixAccountAndReloadData() {
  * pass it in as a parameter. withOnyx guarantees that the value has been read
  * from Onyx because it will not render the AuthScreens until that point.
  * @param {Object} session
+ * @param {string} currentPath
  */
-function setUpPoliciesAndNavigate(session) {
-    Linking.getInitialURL()
-        .then((url) => {
-            if (!url) {
-                return;
-            }
-            const path = new URL(url).pathname;
-            const params = new URLSearchParams(url);
-            const exitTo = params.get('exitTo');
-            const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(url, session.email);
-            const shouldCreateFreePolicy = !isLoggingInAsNewUser
-                        && Str.startsWith(path, Str.normalizeUrl(ROUTES.TRANSITION_FROM_OLD_DOT))
+function setUpPoliciesAndNavigate(session, currentPath) {
```

Comment:
Please update the JSDOC comment to remove the part about `getInitialURL` and maybe add a brief explanation of the `currentPath`.